src/lib/CoreInstance.cc

```diff
@@ -0,0 +1,304 @@
+#include "simeng/CoreInstance.hh"
+
+namespace simeng {
+
+CoreInstance::CoreInstance(int argc, char** argv) {
+  config_ = YAML::Load(DEFAULT_CONFIG);
+  generateCoreModel(argc, argv);
+}
+
+CoreInstance::CoreInstance(int argc, char** argv, std::string configPath) {
```

Comment:
nit: I think the values of argc and argv should be the same as what is passed to main because it is a bit confusing to reason about how `argv[0]` ([here](https://github.com/UoB-HPC/SimEng/pull/247/files#diff-560f14653baa5d07d791071914951617cde676e28ce946ad75f46ffc2ad115dfR82)) in `createProcess` is the executable path and not `./simeng`  unless you look at `main.cc` where the core constructor is invoked as `core(argc-2, argv+2, configPath)`. I think the `if else` clause [here](https://github.com/UoB-HPC/SimEng/pull/247/files#diff-8b47da0924c9ef7fd9025a5091b26a82a6399866b1eee24da80829fb5538870bL65) can be handled inside `generateCoreModel` and then vector a can be passed to `createProcess` in which index 0 is the path to the executable binary,

Reply:
I agree

Reply:
This issue with this is when calling from outside of main, you'll have to "Spoof" the values to match what would have come from main. Maybe a renaming of variables from argc/argv to something more generic (e.g. numCLIArgs, CLIArgs)?

Reply:
So instead why can we simplify the `CoreInstance` constructor to be like:
`CoreInstance(std::string configPath, std::string executablePath, std::vector<std::string> cliArgs)`and then in a `main.cc` context that will be handled externally like it was done in the older version of  `main.cc`.
`if configPath == ""` then load default config (inside constructor) and the same for `executablePath`
This promotes readability and is more convenient for "main.cc" and "non-main.cc" context.
Wdyt?

Reply:
Okay, I'm on board with this. Will make the change.